### DISCUSSION TITLE
Cannot get mkdocs to recognize the caseinsensitive plugin

### DISCUSSION THREAD

**Discussion by ASRFS** [mkdocs/mkdocs#3712] [Q&A] [answered]
I'm trying to use mkdocs to display a body of Markdown documentation that already exists. Unfortunately, some of the pages are `index.md` and others are `Index.md`, and the links to those pages likewise vary. Yes, I know this is all kinds of terrible (https://github.com/mkdocs/mkdocs/issues/1320), but it's also the situation I'm currently in, and it's not feasible to fix it for a bunch of reasons.

I tried to solve it by installing the caseinsensitive plugin (https://github.com/TheMythologist/mkdocs-caseinsensitive-plugin), but when I try `mkdocs serve` it fails with an error stating `Config value 'plugins': The "caseinsensitive" plugin is not installed`.

I've tried a fresh build, but I can't get it to work.

I've looked for other solutions, and found https://github.com/mkdocs/mkdocs/issues/212, but it's not clear from that what I should actually _do_ or whether it will solve my specific problem.

Do you have any suggestions? Thanks for any help you can give.

**Comment by pawamoy** [comment]
> Config value 'plugins': The "caseinsensitive" plugin is not installed

I'm getting the same error. The next step here is to either ask on their repository why this doesn't work, or investigate ourselves. The documentation in their README might be wrong/outdated. I actually don't see any entry for a MkDocs plugin in their project metadata: https://github.com/TheMythologist/mkdocs-caseinsensitive-plugin/blob/main/pyproject.toml.

About #212, I'd recommend the solution mentioned by @waylan, see https://github.com/mkdocs/mkdocs/issues/212#issuecomment-115386890, with the slugify function provided by pymdownx, see for the complete solution: https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#+toc.slugify (maybe do not set `case: lower` though).

**Reply by kamilkrzyskow** [reply]
I didn't try it, but it looks to me like the plugin entry point is `caseinsensitivefiles` not `caseinsensitive`
https://github.com/TheMythologist/mkdocs-caseinsensitive-plugin/blob/23281b35d46e714b48851b6554b0e7b752cd7623/pyproject.toml#L17-L18

**Reply by pawamoy** [reply]
Ah, right, I missed it! Wasn't expecting to find it under `tool.poetry` :sweat_smile:

**Reply by kamilkrzyskow** [reply]
It should be also possible to check the entry points after installation:
1. `pip show mkdocs-caseinsensitive-plugin` open up the `Location` directory
2. `pip show mkdocs-caseinsensitive-plugin -f` look for an `entry_points.txt` file in the list and navigate to it in the `Location`

**Reply by kamilkrzyskow** [reply]
> Ah, right, I missed it! Wasn't expecting to find it under `tool.poetry` 😅

I didn't look, I just did a CTRL+F 😆 Firefox has better search than Chrome, so I'm quite used to using it ✌️

**Comment by ASRFS** [comment, marked as answer]
@kamilkrzyskow , thank you for looking into `caseinsensitiveplugin`; that got the site to build but it did some odd things with links (changed `[PageTitle](relative link)` to just the absolute link as plain text in the file). But it builds!

@pawamoy, thank you as well. The links and hints were enough to get things working. For posterity, the `mkdocs.yml` that builds contains:

```
markdown_extensions:
  - toc:
      slugify: !!python/object/apply:pymdownx.slugs.slugify {kwds: {case: lower}}
```

I found the syntax on https://facelessuser.github.io/pymdown-extensions/faq/#function-references-in-yaml, which was linked from your suggested page of https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#+toc.slugify

I'm not sure I'll stick with `case: lower`. I need to experiment. But now I'm not stuck!

Thank you both very much.
